sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1243,17 +1243,16 @@ case class GpuMultipleContains(input: Expression, searchList: Seq[String])
   override def inputTypes: Seq[AbstractDataType] = Seq(StringType)
 
   override def doColumnar(input: GpuColumnVector): ColumnVector = {
-    assert(searchList.length > 1)
-    val accInit = withResource(Scalar.fromString(searchList.head)) { searchScalar =>
-      input.getBase.stringContains(searchScalar)
+    val targetsBytes = targets.map(t => t.getBytes).toArray
+    val boolCvs = withResource(ColumnVector.fromUTF8Strings(targetsBytes: _*)) { targetsCv =>
+      input.getBase.stringContains(targetsCv)
     }
-    searchList.tail.foldLeft(accInit) { (acc, search) =>
-      val containsSearch = withResource(Scalar.fromString(search)) { searchScalar =>
-        input.getBase.stringContains(searchScalar)
-      }
-      withResource(acc) { _ =>
-        withResource(containsSearch) { _ =>
-          acc.or(containsSearch)
+    closeOnExcept(boolCvs.tail) { _ =>
```

Comment:
This does not work. We will get double frees if there is an exception. `boolCvs.tail` is not updated when the or happens. That is why I wrote the code that I did.

Reply:
We have a complex `safeMap` so that we can have it be safe and at the same time make the resulting code simple.  

https://github.com/NVIDIA/spark-rapids/blob/561068c50f973d5a3751bcc078d41075d13cbbdb/sql-plugin/src/main/scala-2.12/com/nvidia/spark/rapids/implicits.scala#L155-L238

Could we just add a `safeReduce` into implicits.scala instead of trying so hard to make this small piece of code simple?

Reply:
Never mind it is just a helper function `safeReduceAndClose`. We don't need to jump through hoops to make the code simple if inherently what we want to do is not simple.

Reply:
> This does not work. We will get double frees if there is an exception. `boolCvs.tail` is not updated when the or happens. That is why I wrote the code that I did.

Yes you are right, I missed that point and thought an outer `withResource` would lead to double closes. Update back to `withResource` for now.